data/models.py

```diff
@@ -191,3 +191,123 @@ class LandoConnection(models.Model):
+                                         help_text="Workflow that this questionaire is for")
+    questions = models.ManyToManyField(JobQuestion,
+                                       help_text="Questions that are required to create a job")
+
+    def __unicode__(self):
+        return '{} desc:{}'.format(self.id, self.description)
+
+
+class JobAnswerKind(object):
+    """
+    Determines which type of JobAnswer goes with a JobQuestion
+    """
+    STRING = 'string'
+    DDS_FILE = 'dds_file'
+    ITEMS = (
+        (STRING, 'DukeDS File'),
+        (DDS_FILE, 'Text'),
+    )
+
+
+class JobAnswer(models.Model):
+    """
+    Answer to a JobQuestion.
+    """
+    question = models.ForeignKey(JobQuestion, on_delete=models.CASCADE, null=False, related_name='answers')
+    questionnaire = models.ForeignKey(JobQuestionnaire, on_delete=models.CASCADE, null=True, blank=True,
+                                      help_text='When null this is a user editable answer otherwise it is a system answer')
+    index = models.IntegerField(null=True, default=0,
+                                help_text='Used to order array answers')
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, null=False,
+                             help_text='User who created this job answer.')
+    kind = models.CharField(max_length=30, choices=JobAnswerKind.ITEMS,
+                            help_text="Determines child table associated with this answer.")
+
+    def __unicode__(self):
+        return '{} question:{} - index:{}'.format(self.id, self.question.name, self.index)
+
+
+class JobStringAnswer(models.Model):
+    """
+    String value associated with a JobAnswer.
+    """
+    answer = models.OneToOneField(JobAnswer, on_delete=models.CASCADE, primary_key=True, related_name='string_value')
+    value = models.CharField(max_length=255, blank=False, null=False)
+
+    def __unicode__(self):
+        return '{} question:{} - value:{}'.format(self.id, self.answer.question.name, self.value)
+
+
+class JobDDSFileAnswer(models.Model):
+    """
+    DukeDS file keys associated with a JobAnswer.
+    """
+    answer = models.OneToOneField(JobAnswer, on_delete=models.CASCADE, primary_key=True, related_name='dds_file')
+    project_id = models.CharField(max_length=255, blank=False, null=True,
+                                  help_text='uuid from DukeDS for the project containing our file')
+    file_id = models.CharField(max_length=255, blank=False, null=True,
+                               help_text='uuid from DukeDS for the file chosen as this answer')
+    dds_user_credentials = models.ForeignKey(DDSUserCredential, on_delete=models.CASCADE,
+                                             help_text='Credentials with access to this file')
+
+    def __unicode__(self):
+        return '{} question:{} - file_id:{}'.format(self.id, self.answer.question.name, self.file_id)
```

Comment:
Same as above. setting a custom `primary_key` field means that `self.id` doesn't exist.